### PR TITLE
TMDM-11361 : Implement User Management save mdm user information

### PR DESCRIPTION
**Pull request by qcui**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

redesign user management UI save user information

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by qcui** [comment]
code updated
- move EE related file to EE project(PROVISIONING, PROVISIONING.User.administrator,initdb)
- modify DefaultLocalUserDelegator in EE instead of ILocalUser

**Comment by qcui** [comment]
code updated includes
- move initdb related file to be part EE part
- refactor delegator to handle EE part of logic

File: org.talend.mdm.core/src/com/amalto/core/server/DefaultRole.java [hunk 1]
```diff
@@ -95,6 +98,11 @@ public RolePOJOPK removeRole(RolePOJOPK pk) throws XtentisException {
         Collection<ObjectPOJOPK> c = ObjectPOJO.findAllPKs(RolePOJO.class, regex);
         ArrayList<RolePOJOPK> l = new ArrayList<RolePOJOPK>();
         for (ObjectPOJOPK currentObject : c) {
+            if (Util.isEnterprise()) {
```
Comment (pbabin) [inline org.talend.mdm.core/src/com/amalto/core/server/DefaultRole.java]: There's no role in SE, why then this line?

File: org.talend.mdm.core/src/com/amalto/core/delegator/ILocalUser.java [hunk 2]
```diff
@@ -76,11 +76,20 @@ public String getUserXML() {
         return userXml.toString();
     }
 
+    public String getIdentity() {
```
Comment (pbabin) [inline org.talend.mdm.core/src/com/amalto/core/delegator/ILocalUser.java]: Method implementation is exactly the same than getUsername() method

File: org.talend.mdm.core/src/com/amalto/core/util/User.java [hunk 3]
```diff
@@ -134,37 +137,44 @@ public static void parse(String xml, User user) throws Exception {
...
+                }
+                
+                try {
+                    user.setLastSyncTime(new Date(Long.parseLong(Util.getFirstTextNode(result, "//lastsynctime")))); //$NON-NLS-1$
+                } catch (Exception nfe) {
+                    user.setLastSyncTime(null);
+                }
+                user.setEnabled("yes".equals(Util.getFirstTextNode(result, "//enabled"))); //$NON-NLS-1$ //$NON-NLS-2$
+                user.setHomePage(Util.getFirstTextNode(result, "//homepage")); //$NON-NLS-1$
+                user.setUniverse(Util.getFirstTextNode(result, "//universe")); //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (qcui) [reply org.talend.mdm.core/src/com/amalto/core/util/User.java]: right now this part should support CE & EE users management, first part is for EE, second is for CE

File: org.talend.mdm.webapp.core/src/main/java/com/amalto/webapp/core/util/Util.java [hunk 4]
```diff
@@ -839,9 +841,21 @@ public static void filterAuthViews(Map<String, String> viewMap) throws Exception
 
     public static String getDefaultLanguage() throws Exception {
         String defaultLanguage = ""; //$NON-NLS-1$
-        String userName = LocalUser.getLocalUser().getUsername();
-        WSItemPK itemPK = new WSItemPK(new WSDataClusterPK(DATACLUSTER_PK), PROVISIONING_CONCEPT, new String[] { userName });
-        if (userName != null && userName.length() > 0) {
+        String identity = LocalUser.getLocalUser().getIdentity();
+        if (com.amalto.core.util.Util.isEnterprise()) {
```
[the remark this answers is not in the document]
Comment (qcui) [reply org.talend.mdm.webapp.core/src/main/java/com/amalto/webapp/core/util/Util.java]: CE version user has no id information.

File: org.talend.mdm.core/src/com/amalto/core/server/DefaultRole.java (hunk 1 above)
Comment (qcui) [reply org.talend.mdm.core/src/com/amalto/core/server/DefaultRole.java]: I think you are right.

File: org.talend.mdm.webapp.welcomeportal/src/main/java/org/talend/mdm/webapp/welcomeportal/server/actions/WelcomePortalAction.java [hunk 5]
```diff
@@ -259,6 +263,16 @@ public void savePortalConfig(PortalProperties config) throws ServiceException {
...
+            if (com.amalto.core.util.Util.isEnterprise() && parsedUser.getRoleNames() != null) {
+                Set<String> roleNames = new HashSet<String>();
+                for (String role : parsedUser.getRoleNames()) {
+                    if (role != null && role.length() > 7 && SYSTEM_PREFIX.equals(role.substring(0, 7))) {
+                        continue;
+                    }
+                    roleNames.add(role);
+                }
+                parsedUser.setRoleNames(roleNames);
+            }
```
Comment (pbabin) [inline org.talend.mdm.webapp.welcomeportal/src/main/java/org/talend/mdm/webapp/welcomeportal/server/actions/WelcomePortalAction.java]: There is a problem here

**Comment by qcui** [comment]
Modified code according to review comment. and created two tickets to handling remain work.
TMDM-11432 User unique identifier in SE&EE versions
TMDM-11433 General refactor user information handling process

File: org.talend.mdm.webapp.welcomeportal/src/main/java/org/talend/mdm/webapp/welcomeportal/server/actions/WelcomePortalAction.java [hunk 6]
```diff
@@ -259,6 +263,16 @@ public void savePortalConfig(PortalProperties config) throws ServiceException {
                 return;
             }
             User parsedUser = User.parse(user.getUserXML());
+            if (com.amalto.core.util.Util.isEnterprise() && parsedUser.getRoleNames() != null) {
+                Set<String> roleNames = new HashSet<String>();
+                for (String role : parsedUser.getRoleNames()) {
+                    if (role != null && role.length() > 7 && SYSTEM_PREFIX.equals(role.substring(0, 7))) {
```
Comment (htyin) [inline org.talend.mdm.webapp.welcomeportal/src/main/java/org/talend/mdm/webapp/welcomeportal/server/actions/WelcomePortalAction.java]: We can use `role.startsWith(SYSTEM_PREFIX)`, right?

File: org.talend.mdm.core/src/com/amalto/core/server/DefaultRole.java [hunk 7]
```diff
@@ -95,6 +98,9 @@ public RolePOJOPK removeRole(RolePOJOPK pk) throws XtentisException {
         Collection<ObjectPOJOPK> c = ObjectPOJO.findAllPKs(RolePOJO.class, regex);
         ArrayList<RolePOJOPK> l = new ArrayList<RolePOJOPK>();
         for (ObjectPOJOPK currentObject : c) {
+            if (currentObject.getIds().length > 0 && currentObject.getIds()[0].length() > 7 && SYSTEM_PREFIX.equals(currentObject.getIds()[0].substring(0, 7)) ) {
```
Comment (htyin) [inline org.talend.mdm.core/src/com/amalto/core/server/DefaultRole.java]: `currentObject.getIds()[0].startWith(SYSTEM_PREFIX )` can be used here, right?

**Pull request merged**